conn.go

```diff
@@ -79,6 +100,25 @@ func (c *Conn) Write(b []byte) (err error) {
 	return nil
 }
 
+func (c *Conn) FlushWriter() (err error) {
+	bufWriter, ok := c.writer.(*bufio.Writer)
+	if ok {
+		c.wmtx.Lock()
+		defer c.wmtx.Unlock()
```

Comment:
It's not a big deal but defer calls take a bit of extra cpu. As we're only locking a single line, we can easily avoid the defer.

```
c.wmtx.Lock()
err = flushWriter(bufWriter)
c.wmtx.Unlock()

return err
```